src/Functions/custom/New-AzFunctionAppPlan.ps1

```diff
@@ -0,0 +1,175 @@
+        {
+            $errorMessage = "MaximumWorkerCount is only supported for Elastic Premium (EP) plans."
+            $exception = [System.InvalidOperationException]::New($errorMessage)
+            ThrowTerminatingError -ErrorId "MaximumWorkerCountIsOnlySupportedForElasticPremiumPlan" `
+                                  -ErrorMessage $errorMessage `
+                                  -ErrorCategory ([System.Management.Automation.ErrorCategory]::InvalidOperation) `
+                                  -Exception $exception
+        }
+
+        if ($MaximumWorkerCount -lt $MinimumWorkerCount)
+        {
+            $errorMessage = "MinimumWorkerCount '$($MinimumWorkerCount)' cannot be less than '$($MaximumWorkerCount)'."
+            $exception = [System.InvalidOperationException]::New($errorMessage)
+            ThrowTerminatingError -ErrorId "MaximumWorkerCountIsOnlySupportedForElasticPremiumPlan" `
+                                  -ErrorMessage $errorMessage `
+                                  -ErrorCategory ([System.Management.Automation.ErrorCategory]::InvalidOperation) `
+                                  -Exception $exception
+        }
+
+        # Validate location for a Premium plan
+        $Location = $Location.Trim()
+        $availableLocations = @(Get-AzFunctionAppAvailableLocation -OSType $WorkerType -PlanType Premium | ForEach-Object { $_.Name })
+        if (-not ($availableLocations -contains $Location))
+        {
+            $locationOptions = $availableLocations -join ", "
+            $errorMessage = "Location is invalid. Currently supported locations are: $locationOptions"
+            $exception = [System.InvalidOperationException]::New($errorMessage)
+            ThrowTerminatingError -ErrorId "LocationIsInvalid" `
+                                  -ErrorMessage $errorMessage `
+                                  -ErrorCategory ([System.Management.Automation.ErrorCategory]::InvalidOperation) `
+                                  -Exception $exception
+        }
+
+        $servicePlan = New-Object -TypeName Microsoft.Azure.PowerShell.Cmdlets.Functions.Models.Api20180201.AppServicePlan
+
+        # Plan settings
+        $servicePlan.SkuTier = $tier
+        $servicePlan.SkuName = $Sku
+        $servicePlan.Location = $Location
+        $servicePlan.Tag = $Tag
+        $servicePlan.Reserved = ($WorkerType -eq "Linux")
+
+        if ($MinimumWorkerCount -gt 0)
+        {
+            $servicePlan.Capacity = $MinimumWorkerCount
+        }
+
+        if ($MaximumWorkerCount -gt 0)
+        {
+            $servicePlan.MaximumElasticWorkerCount = $MaximumWorkerCount
+        }
+
+        $null = $PSBoundParameters.Add("AppServicePlan", $servicePlan)
+        $null = $PSBoundParameters.Add("ErrorAction", "Stop")
+
+        try
+        {
+            $createdPlan = Az.Functions.internal\New-AzFunctionAppPlan @PSBoundParameters
+
+            if ($createdPlan)
```

Comment:
This is what I am talking about  tiwht the 'Asjob' setting - the above cmdlet will return a Job, which will cause the rest of the cmdlet to fail (and also, not return a job)

Reply:
Agsin, we should either remove the parameters (AsJob, NoWait) or fix for preview